### PR TITLE
USM total hits TLS library tag

### PR DESCRIPTION
**Pull request by DanielLavie**
### What does this PR do?
Introduces a TLS library label to the USM HTTP total hits metric

### Motivation
When examining the HTTP total hits metric (e.g. `datadog.network_tracer.usm.http.total_hits`), we aim to gain insights into the usage of a particular TLS library, if any, in order to enhance our understanding and improve our debugging capabilities

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/http/telemetry.go [hunk 1]
```diff
@@ -96,8 +102,18 @@ func (t *Telemetry) Count(tx Transaction) {
 		t.hits5XX.Add(1)
 	}
 
-	if isEncrypted(tx) {
-		t.totalHitsEncrypted.Add(1)
+	switch tx.StaticTags() {
+	case GnuTLS:
+		t.totalHitsGnuTLS.Add(1)
+		return
```
Comment (guyarb) [inline pkg/network/protocols/http/telemetry.go:108]: the return is redundant in every case, as go's default mechanism is NOT fallthrough

File: pkg/network/protocols/http/telemetry.go [hunk 2]
```diff
@@ -63,7 +66,10 @@ func NewTelemetry(protocol string) *Telemetry {
 
 		// these metrics are also exported as statsd metrics
 		totalHitsPlain:         metricGroup.NewCounter("total_hits", "encrypted:false", libtelemetry.OptStatsd, libtelemetry.OptPayloadTelemetry),
-		totalHitsEncrypted:     metricGroup.NewCounter("total_hits", "encrypted:true", libtelemetry.OptStatsd, libtelemetry.OptPayloadTelemetry),
+		totalHitsGnuTLS:        metricGroup.NewCounter("total_hits", "encrypted:true", "tls_library:gnutls", libtelemetry.OptStatsd, libtelemetry.OptPayloadTelemetry),
+		totalHitsOpenSLL:       metricGroup.NewCounter("total_hits", "encrypted:true", "tls_library:openssl", libtelemetry.OptStatsd, libtelemetry.OptPayloadTelemetry),
+		totalHitsJavaTLS:       metricGroup.NewCounter("total_hits", "encrypted:true", "tls_library:java", libtelemetry.OptStatsd, libtelemetry.OptPayloadTelemetry),
+		totalHitsGoTLS:         metricGroup.NewCounter("total_hits", "encrypted:true", "tls_library:go", libtelemetry.OptStatsd, libtelemetry.OptPayloadTelemetry),
```
Comment (guyarb) [inline pkg/network/protocols/http/telemetry.go:72]: I think we should extend (can be in a different PR) the other metrics as well to have tls tag

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/protocols/http/telemetry.go (hunk 2 above)
Comment (DanielLavie) [reply pkg/network/protocols/http/telemetry.go]: Agreed:
https://datadoghq.atlassian.net/browse/USMON-618

**Review by guyarb — APPROVED** [review]

**Review by ISauve — APPROVED** [review]
👍  for WKIT files

File: pkg/network/protocols/http/telemetry.go [hunk 3]
```diff
@@ -96,12 +102,18 @@ func (t *Telemetry) Count(tx Transaction) {
...
-		return
+	switch tx.StaticTags() {
+	case GnuTLS:
+		t.totalHitsGnuTLS.Add(1)
+	case OpenSSL:
+		t.totalHitsOpenSLL.Add(1)
+	case Java:
+		t.totalHitsJavaTLS.Add(1)
+	case Go:
+		t.totalHitsGoTLS.Add(1)
```
Comment (guyarb) [inline pkg/network/protocols/http/telemetry.go:113]: those tags do not exist on a windows build

**Review by Yumasi — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Review by amitslavin — APPROVED** [review]
